zebra-network/src/protocol/external/codec.rs

```diff
@@ -686,4 +694,71 @@ mod tests {
         assert_eq!(format!("{:?}", decode_state),
                    "DecodeState::Body { body_len: 43, command: \"v�ion\\u{0}\\u{0}\\u{0}\\u{0}\\u{0}\", checksum: Sha256dChecksum(\"bafaa2e3\") }");
     }
+
+    #[test]
+    fn max_msg_size_round_trip() {
+        use std::sync::Arc;
+        let mut rt = Runtime::new().unwrap();
+
+        // make tests with a Tx message
+        let tx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap();
```

Comment:
TIOLI: I recently added a helper to make working with deserialize a little easier. This version reorders the arguments to make it a method call which enables auto-deref. That way you don't have to do the whole `&vec[..]` rigmarole.

```suggestion
        let tx = zebra_test::vectors::DUMMY_TX1
            .zcash_deserialize_into()
            .unwrap();
```

Also because it can use method call syntax you its easier to let type inference take over, which works fine in this case. In the future if you end up getting errors about the compiler being unable to infer the type you can always specify it similar to `collect`.

```rust
        let tx = zebra_test::vectors::DUMMY_TX1
            .zcash_deserialize_into::<Transaction>()
            .unwrap();
```

Reply:
Nice one, applied at https://github.com/ZcashFoundation/zebra/pull/645/commits/d60ccac5f7902a4f0ce24c1f6de0b6f71c95a7ac 

Sorry i was not aware of this, will use it in the future.